lightning/src/ln/chan_utils.rs

```diff
@@ -217,87 +217,100 @@ pub fn derive_public_key<T: secp256k1::Signing>(secp_ctx: &Secp256k1<T>, per_com
 	let rev_append_commit_hash_key = {
 		let mut sha = Sha256::engine();
-		sha.input(&revocation_base_point.serialize());
-		sha.input(&per_commitment_point.serialize());
+		sha.input(&countersignatory_revocation_base_point.serialize());
+		sha.input(&broadcaster_per_commitment_point.serialize());
 
 		Sha256::from_engine(sha).into_inner()
 	};
 	let commit_append_rev_hash_key = {
 		let mut sha = Sha256::engine();
-		sha.input(&per_commitment_point.serialize());
-		sha.input(&revocation_base_point.serialize());
+		sha.input(&broadcaster_per_commitment_point.serialize());
+		sha.input(&countersignatory_revocation_base_point.serialize());
 
 		Sha256::from_engine(sha).into_inner()
 	};
 
-	let mut part_a = revocation_base_point.clone();
-	part_a.mul_assign(&secp_ctx, &rev_append_commit_hash_key)?;
-	let mut part_b = per_commitment_point.clone();
-	part_b.mul_assign(&secp_ctx, &commit_append_rev_hash_key)?;
-	part_a.combine(&part_b)
+	let mut countersignatory_contrib = countersignatory_revocation_base_point.clone();
+	countersignatory_contrib.mul_assign(&secp_ctx, &rev_append_commit_hash_key)?;
+	let mut broadcaster_contrib = broadcaster_per_commitment_point.clone();
+	broadcaster_contrib.mul_assign(&secp_ctx, &commit_append_rev_hash_key)?;
+	countersignatory_contrib.combine(&broadcaster_contrib)
 }
 
 /// The set of public keys which are used in the creation of one commitment transaction.
 /// These are derived from the channel base keys and per-commitment data.
 ///
+/// A broadcaster key is provided from potential broadcaster of the computed transaction.
+/// A countersignatory key is coming from a protocol participant unable to broadcast the
+/// transaction.
+///
 /// These keys are assumed to be good, either because the code derived them from
 /// channel basepoints via the new function, or they were obtained via
 /// PreCalculatedTxCreationKeys.trust_key_derivation because we trusted the source of the
 /// pre-calculated keys.
 #[derive(PartialEq, Clone)]
 pub struct TxCreationKeys {
-	/// The per-commitment public key which was used to derive the other keys.
+	/// The broadcaster's per-commitment public key which was used to derive the other keys.
 	pub per_commitment_point: PublicKey,
-	/// The revocation key which is used to allow the owner of the commitment transaction to
-	/// provide their counterparty the ability to punish them if they broadcast an old state.
-	pub revocation_key: PublicKey,
-	/// A's HTLC Key
-	pub a_htlc_key: PublicKey,
-	/// B's HTLC Key
-	pub b_htlc_key: PublicKey,
-	/// A's Payment Key (which isn't allowed to be spent from for some delay)
-	pub a_delayed_payment_key: PublicKey,
+	/// The broadcaster's revocation key which is used to allow the broadcaster of the commitment
+	/// transaction to provide their counterparty the ability to punish them if they broadcast
+	/// an old state.
+	pub broadcaster_revocation_key: PublicKey,
```

Comment:
I think I'd be happier if this just said "revocation_key" to match "per_commitment_point" - its a private key that requires private data from both sides, and really only the countersignatory can ever learn the private key part, but its awkward to suggest its owned by either side. Note that this change would effect a few other cases of "revocation_key" in this file.

Reply:
Good call, shared keys are hard to attribute ownership. Corrected and normally other references spread over.